authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationHDFSPlugin.java

```diff
@@ -278,88 +696,91 @@ public List<AuthorizationSecurableObject> translateOwner(MetadataObject gravitin
-                        NameIdentifier.of(
-                            metalake, nsMetadataObj.level(0), "default" /*Hive default schema*/));
-            String defaultSchemaLocation = schema.properties().get(HiveConstants.LOCATION);
-            locationPath = pattern.matcher(defaultSchemaLocation).replaceAll("");
-          }
-        }
-        break;
-      case FILESET:
-        FilesetDispatcher filesetDispatcher = GravitinoEnv.getInstance().filesetDispatcher();
-        NameIdentifier identifier = getObjectNameIdentifier(metadataObject);
-        Fileset fileset = filesetDispatcher.loadFileset(identifier);
+  @Override
+  public Boolean onMetadataUpdated(MetadataObjectChange... changes) throws RuntimeException {
+    for (MetadataObjectChange change : changes) {
+      if (change instanceof MetadataObjectChange.RenameMetadataObject) {
+        MetadataObject metadataObject =
+            ((MetadataObjectChange.RenameMetadataObject) change).metadataObject();
+        MetadataObject newMetadataObject =
+            ((MetadataObjectChange.RenameMetadataObject) change).newMetadataObject();
         Preconditions.checkArgument(
-            fileset != null, String.format("Fileset %s is not found", identifier));
-        String filesetLocation = fileset.storageLocation();
+            metadataObject.type() == newMetadataObject.type(),
+            "The old and new metadata object type must be equal!");
+        if (metadataObject.type() == MetadataObject.Type.METALAKE) {
+          // Rename the metalake name
+          this.metalake = newMetadataObject.name();
+          // Did not need to update the Ranger policy
+          continue;
+        } else if (metadataObject.type() == MetadataObject.Type.CATALOG) {
+          // Did not need to update the Ranger policy
+          continue;
+        }
+        List<AuthorizationMetadataObject> oldAuthzMetadataObjects =
+            translateMetadataObject(metadataObject);
+        List<AuthorizationMetadataObject> newAuthzMetadataObjects =
+            translateMetadataObject(newMetadataObject);
         Preconditions.checkArgument(
-            filesetLocation != null, String.format("Fileset %s location is not found", identifier));
-        locationPath = pattern.matcher(filesetLocation).replaceAll("");
-        break;
-      default:
-        throw new AuthorizationPluginException(
-            "The metadata object type %s is not supported in the RangerAuthorizationHDFSPlugin",
-            metadataObject.type());
+            oldAuthzMetadataObjects.size() == newAuthzMetadataObjects.size(),
+            "The old and new metadata objects size must be equal!");
+        for (int i = 0; i < oldAuthzMetadataObjects.size(); i++) {
+          AuthorizationMetadataObject oldAuthMetadataObject = oldAuthzMetadataObjects.get(i);
+          AuthorizationMetadataObject newAuthzMetadataObject = newAuthzMetadataObjects.get(i);
+          if (oldAuthMetadataObject.equals(newAuthzMetadataObject)) {
```

Comment:
PathBasedMetadataObject has not overwrites `equals`, so there may be problems to use equals

```java
    PathBasedMetadataObject pathBasedMetadataObject1 =
        new PathBasedMetadataObject("parent", "name", "path", PathBasedMetadataObject.Type.PATH);
    pathBasedMetadataObject1.validateAuthorizationMetadataObject();


    PathBasedMetadataObject pathBasedMetadataObject2 =
        new PathBasedMetadataObject("parent", "name", "path", PathBasedMetadataObject.Type.PATH);
    pathBasedMetadataObject2.validateAuthorizationMetadataObject();

    boolean result = pathBasedMetadataObject1.equals(pathBasedMetadataObject2); // alreays false
    System.out.println(result);
```